packages/perseus/src/components/zoomable.tsx

```diff
@@ -261,7 +265,10 @@ class Zoomable extends React.Component<Props, State> {
             : {};
 
         // Do a fancy little slide as we fade the contents in the first time.
-        const translateOffset = visible ? "" : " translate(0, 8px)";
+        const shouldSlide = !disableEntranceAnimation && !visible;
+        const translateOffset = shouldSlide ? " translate(0, 8px)" : "";
+        const opacity = shouldSlide ? 0 : 1;
+
```

Comment:
suggestion: I find this (and lines 272-275) a little hard to follow with all the boolean logic and ternaries. It might be nice to pull this code out into a function that could be tested. IMO everything between lines 252 and 275 could go into that function. Calling it would look something like this:

```
const transitionStyle = computeTransitionStyle({
  visible,
  zoomed,
  scale,
  animateHeight,
  disableEntranceAnimation,
});
```

I think I'd find reading the tests for that function easier than trying to work through all the possibilities in my head.